src/renderer/atlas/BackendD3D.cpp

```diff
@@ -1627,9 +1628,19 @@ void BackendD3D::_drawCursorBackground(const RenderingPayload& p)
             static_cast<u16>(p.s->font->cellSize.x * (x1 - x0)),
             p.s->font->cellSize.y,
         };
-        const auto isInverted = cursorColor == 0xffffffff;
-        const auto background = isInverted ? bg ^ 0xc0c0c0 : cursorColor;
-        const auto foreground = isInverted ? 0 : bg;
+        auto background = cursorColor;
+        auto foreground = bg;
+
+        if (cursorColor == 0xffffffff)
+        {
+            background = bg ^ 0xffffff;
+            foreground = 0xffffffff;
+        }
+
+        // The legacy console used to invert colors by just doing `bg ^ 0xc0c0c0`. This resulted
+        // in a minimum squared distance of just 0.029195 across all possible color combinations.
+        background = ColorFix::GetPerceivableColor(background, bg, 0.25f * 0.25f);

```

Comment:
The minimum distance given to `GetPerceivableColor` is the squared distance. That way we don't need to do any `sqrt()` inside it (which is very slow relatively speaking). The background of the cursor gets half the min. distance the text gets, because I noticed that the background color really just doesn't need it.

Note to self: Investigate `"\e[48;2;128;128;127m"` distance.